hibernate-core/src/main/java/org/hibernate/loader/ast/internal/SingleIdLoadPlan.java

```diff
@@ -137,23 +131,32 @@ public T load(
 			);
 		}
 		assert offset == jdbcParameters.size();
-		final QueryOptions queryOptions = new SimpleQueryOptions( lockOptions, readOnly );
 		final Callback callback = new CallbackImpl();
+		final SubselectFetch.RegistrationHandler subselectRegistrationHandler = SubselectFetch.createRegistrationHandler(
+				session.getPersistenceContextInternal().getBatchFetchQueue(),
+				sqlAst,
+				jdbcParameters,
+				jdbcParameterBindings
+		);
```

Comment:
It would be great to avoid having to pass the `sqlAst` here. If you look at the `SubselectFetch.createRegistrationHandler` implementation, you can see it's used for 2 things:
1. checking whether the query has at least 1 root
2. passing the first `TableGroup` to the `StandardRegistrationHandler` constructor

Regarding point 2., the constructor parameter is never actually used - so we can deprecate this constructor and create one without the `TableGroup` parameter.

As for point 1., in `SingleIdLoadPlan` we always know the generated query has at least one root (we can be extra sure by adding an `assert !sqlAst.getQuerySpec().getFromClause().getRoots().isEmpty()` in the constructor) - so we don't really need to pass it for this check. I suggest creating a third signature of the `createRegistrationHandler` with the same parameters but no `TableGroup`, and deprecate the one with it.

Reply:
@mbladel You might have overlooked something, because as far as I can see the QuerySpec (from sqlAst) is used for the generation of the SubselectFetch query. The original query is copied into the subselect query 'in (... original query ... )'.

It might be possible to generate de subselection query by other means (ie recreate the original queryspec while subselecting)? Otherwise I do not think we can do without at least the original queryspec. What is your opinion on this?

Reply:
You're right, and e.g. `EntityBatchLoaderArrayParam` already stores the SQL AST to use in subselect registration handlers, so we might do the same here.